array-manipulation-1/src/compact.js

```diff
@@ -0,0 +1,15 @@
+/* exported compact */
+/* Make storage space
+- Go through each value of the array
+-if the value holds false, delete the output into the storage
+
+*/
+function compact(array) {
+  var hipHipArray = [];
+  for (var i = 0; i < array.length; i++) {
+    if (!!array[i] === false) {
+      delete array[i];
```

Comment:
Oh no, this is making changes to the original array, that's bad. You don't need to do anything with these values, you are just collecting the values that are equivalent to the `true` boolean.